crates/typst/src/doc.rs

```diff
@@ -207,7 +201,9 @@ impl Frame {
     where
         I: IntoIterator<Item = (Point, FrameItem)>,
     {
-        Arc::make_mut(&mut self.items).splice(0..0, items);
+        for (i, item) in items.into_iter().enumerate() {
+            self.items.insert(i, item);
+        }
```

Comment:
It shows no difference, there just not enough splicing to get gains, but I think it would be better to wait for ecow to have `splice` as a method (I'll see if I can open a PR tomorrow) instead of this, I'll revert the change today.